### PR TITLE
test: patch existing CSV for a hotfix

### PR DESCRIPTION
**Pull request by jeyaramashok**
**Description of the change:**

This is a e2e test case for hot fixing a CSV. In this scenario an existing CSV is patched directly to update the deployment spec.

**Motivation for the change:**

hot fix scenario described in https://github.com/operator-framework/operator-lifecycle-manager/issues/1774

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Review by ankitathomas — APPROVED** [review]
/ok-to-test

File: test/e2e/csv_e2e_test.go [hunk 1]
```diff
@@ -2664,7 +2664,136 @@ var _ = Describe("CSV", func() {
...
+		// Fetch the current csv
+		fetchedCSV, err := fetchCSV(crc, csv.Name, testNamespace, csvSucceededChecker)
+		require.NoError(GinkgoT(), err)
+
+		// Update csv with modified deployment spec
+		fetchedCSV.Spec.InstallStrategy.StrategySpec = strategyNew
+
+		// Update the current csv
+		_, err = crc.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Update(context.TODO(), fetchedCSV, metav1.UpdateOptions{})
+		require.NoError(GinkgoT(), err)
```
Comment (awgreene) [inline test/e2e/csv_e2e_test.go:2785]: Could this all be done in an Eventually() command? As it is, the update command may occasionally fail if another resource modifies the CSV before it is fetched.

**Review by awgreene — CHANGES_REQUESTED** [review]
One change request, looks good otherwise.

File: test/e2e/csv_e2e_test.go [hunk 2]
```diff
@@ -2664,7 +2664,136 @@ var _ = Describe("CSV", func() {
 		err = waitForDeploymentToDelete(GinkgoT(), c, strategy.DeploymentSpecs[0].Name)
 		require.NoError(GinkgoT(), err)
 	})
+	It("update deployment spec in an existing CSV for a hotfix", func() {
```
Comment (timflannagan) [inline test/e2e/csv_e2e_test.go:2667]: It looks like you may need to rebase against master. When I look at the raw view of this file, that testify/require package is present in the list of package imports, but not present in this same file in master: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/csv_e2e_test.go#L3-L35, which explains why the e2e tests are failing due `test/e2e/csv_e2e_test.go:3045:3: undefined: require`.

File: test/e2e/csv_e2e_test.go (hunk 1 above)
Comment (jeyaramashok) [reply test/e2e/csv_e2e_test.go]: sorry I missed this msg, I have update to include eventually() block https://github.com/operator-framework/operator-lifecycle-manager/pull/1817/files#diff-e94bc9e5ebd2b6db20f851288184630432208a5a772cf0515935707ad3660799R3034

File: test/e2e/csv_e2e_test.go (hunk 2 above)
Comment (jeyaramashok) [reply test/e2e/csv_e2e_test.go:2918]: rebased the PR with latest on master

Comment (jeyaramashok) [reply test/e2e/csv_e2e_test.go:2918]: I verified that the test passes on my local machine.

**Pull request merged**